### ISSUE TITLE
Event Endzeit wird angezeigt

### ISSUE THREAD

**Issue by xyzjuhe** [contao/contao#6608]
### Affected version(s)

Version 4.13.35

### Description

Nach dem Update auf 4.13.35 wird die Endzeit angezeigt, auch wenn das Feld leer ist.
Unter 4.13.3 funktioniert es bestens.

**Comment by fritzmg** [comment]
I cannot confirm the issue with any of the standard `event_*` templates. Are you using a custom `event_*` template?

**Comment by xyzjuhe** [comment]
I actually need my own tempalte.
Sorry that I didn't mention that.
Attached you will find my template. Curiously, it worked wonderfully so far...

Kind regards
Aitana

**Comment by fritzmg** [comment]
You need to log in to Github an post your template here directly - instead of replying via email.

**Comment by xyzjuhe** [comment]
Thank you for your hint. I'm not very familiar with Github yet.


```php
<div class="event layout_teaser<?= $this->classList ?>" itemscope itemtype="http://schema.org/Event">

		<?php
		// date_default_timezone_set('Europe/Berlin');
		// $loc_de = setlocale(LC_ALL, 'de_DE', 'de_DE@euro', 'de_DE.UTF-8', 'deu_deu', 'DEU', 'de', 'ge'); 
		
		$array = ["Januar", "Februar", "März", "April","Mai","Juni", "Juli", "August", "September", "Oktober", "November", "Dezember"];
		
		
		// Starttag und Endtag vergleichen
		if($this->startDate && $this->endDate) 
		{
			$monatsIndex = (int)date("n",$this->startDate)-1; // Startmonat
			$start[1] = $array[$monatsIndex];

			$monatsIndex = (int)date("n",$this->endDate)-1; // Endemonat
			$ende[1] = $array[$monatsIndex];
	
			$start[0] = date("d",$this->startDate); // Starttag
			//$start[1] = date("F",$this->startDate); // Startmonat
			$start[2] = date("Y",$this->startDate); // Startjahr
			$ende[0] = date("d",$this->endDate); // Endetag
			//$ende[1] = date("F",$this->endDate); // Endemonat
			$ende[2] = date("Y",$this->endDate); // Endejahr
			if($start[2] == $ende[2]) 
		{
			// gleiches Jahr
			$temp[0] = "";
			$temp[1] = $ende[2];
		}
			else
		{
			// unterschiedliches Jahr
			$temp[0] = $start[2];
			$temp[1] = $ende[2];
		}
			if($start[1] == $ende[1]) 
			{
			// gleicher Monat
			$temp[1] = $ende[1]." ".$temp[1];
		}
		  else
		{
		// unterschiedlicher Monat
			$temp[0] = $start[1]." ".$temp[0];
			$temp[1] = $ende[1]." ".$temp[1];
		}
			// if($start[0] == $ende[0]) 
			// {
			// gleicher Tag
			// $temp[1] = $ende[0].". ".$temp[1];
			// }
			// else
		{
			// unterschiedlicher Tag
			$temp[0] = $start[0].". ".$temp[0];
			$temp[1] = $ende[0].". ".$temp[1];
		}
			$anzeigetag = $temp[0]." – ".$temp[1];
		}
		else
		{
			// Endtag ist nicht gesetzt
			//$anzeigetag = date("l, j. F Y",$this->startDate);
		}

		?>


		<?php if ($this->endDate == true ): ?>
		<!--<?php echo($array[$monatsIndex]); ?>-->

					<span class="datum"><?php echo $anzeigetag; ?></span>
					<?php if ($this->addTime): ?>
					<?php if ($this->startTime != $this->endTime): ?> 
					<span class="uhrzeit"><?php echo $this->parseDate("H:i", $this->startTime); ?> –  
					<?php echo $this->parseDate("H:i", $this->endTime); ?> Uhr</span>
					<?php endif; ?> 
 
					<?php if ($this->startTime == $this->endTime): ?> 
					<span class="uhrzeit"><?php echo $this->parseDate("H:i", $this->startTime); ?> Uhr</span>
					<?php endif; ?>
					<?php endif; ?>
				


		<?php else: ?>
		
		<?php global $objPage; ?>
		<?php /* <time class="time"><?php echo $this->parseDate("l, j. F Y", $this->startTime); ?>&nbsp;&nbsp;&nbsp; */ ?>
			<span class="datum"><?php $currentDate = DateTime::createFromFormat($objPage->dateFormat, $this->date); echo $this->parseDate("D, d. F Y", $currentDate->getTimestamp()); ?></span>
				<?php if ($this->addTime): ?>
				<?php if ($this->startTime != $this->endTime): ?>
				<span class="uhrzeit"><?php echo $this->parseDate("H:i", $this->startTime); ?> – 
				<?php echo $this->parseDate("H:i", $this->endTime); ?> Uhr</span>
				<?php endif; ?> 

				<?php if ($this->startTime == $this->endTime): ?>
				<span class="uhrzeit"><?php echo $this->parseDate("H:i", $this->startTime); ?> Uhr</span>
				<?php endif; ?>
				<?php endif; ?>
			</span>
	
		<?php endif;?>
	
	<?php if ($this->location): ?>
	 <span class="location" itemprop="location" itemscope itemtype="http://schema.org/Place">
		<span itemprop="name"><?= $this->location ?></span>
			<?php if ($this->address): ?>
			(<span itemprop="address" itemscope itemtype="http://schema.org/PostalAddress"><?= $this->address ?></span>)
			<?php endif; ?>
		</span>
	<?php endif; ?>




  <?php if ($this->hasDetails): ?>
    <h2 itemprop="name"><a href="<?= $this->href ?>" title="<?= $this->title ?> (<?php if ($this->day): ?><?= $this->day ?>, <?php endif; ?><?= $this->date ?><?php if ($this->time): ?> <?= $this->time ?><?php endif; ?>)"<?= $this->target ?> itemprop="url"><?= $this->link ?></a></h2>
  <?php else: ?>
    <h2 itemprop="name"><?= $this->title ?></h2>
  <?php endif; ?>

  


  <div class="ce_text block" itemprop="description">
    <?php if (!$this->addBefore): ?>
      <?= $this->teaser ?>
    <?php endif; ?>

    <?php if ($this->addImage): ?>
      <?php $this->insert('image', $this->arrData); ?>
    <?php endif; ?>

    <?php if ($this->addBefore): ?>
      <?= $this->teaser ?>
    <?php endif; ?>
  </div>

  <?php if ($this->hasDetails): ?>
    <p class="more"><a href="<?= $this->href ?>" title="<?= $this->readMore ?>"<?= $this->target ?> itemprop="url"><?= $this->more ?><span class="invisible"> <?= $this->link ?></span></a></p>
  <?php endif; ?>

</div>

```

**Comment by fritzmg** [comment]
@ausi I ran into the same issue actually. Previously we always checked via `$this->startTime !== $this->endTime` in a template whether an event has an end time or not. Now that we override the `endTime` with `23:59:59` when the event has no end time that does not work anymore.

@asdfasdfasa you can change your `$this->startTime != $this->endTime` checks to `$this->begin !== $this->end` instead I think.

@ausi do you think we can change this back to the previous behavior somehow? I think the `$this->startTime !== $this->endTime` check is fairly common in order to check whether an event has an end time or not in the template (I think there is no other obvious way to do it, except using `$this->begin` and `$this->end`.

**Comment by ausi** [comment]
> do you think we can change this back to the previous behavior somehow?

We could add a new field like `endTimeForRealFinalV2` and use it here https://github.com/contao/contao/blob/56c8fe3e5693a333616d94f3911501fb3a34fcf7/calendar-bundle/src/Resources/contao/classes/Events.php#L312 and here https://github.com/contao/contao/blob/56c8fe3e5693a333616d94f3911501fb3a34fcf7/calendar-bundle/src/Resources/contao/modules/ModuleEventlist.php#L205

**Comment by fritzmg** [comment]
> We could add a new field like `endTimeForRealFinalV2`

😁 

May be `virtualEndTime`?

**Comment by ausi** [comment]
> May be `virtualEndTime`?

How about `validEndTime` or `finalEndTime`?

**Comment by leofeyer** [comment]
> you can change your `$this->startTime != $this->endTime` checks to `$this->begin !== $this->end` instead I think.

Does this work?

**Comment by fritzmg** [comment]
This is how I fixed it in our project after updating to 4.13.35. Not sure if this will work in every case.

**Comment by leofeyer** [comment]
As discussed in the Contao call, we want to fix it as outlined in https://github.com/contao/contao/issues/6608#issuecomment-1845533141 and the variable should be called `effectiveEndTime`.
